Added tests for CreateUserAccount, BulkEnrollment and UpdateUserAccount API views

OmarIthawi opened the pull request:
**Note:** The Bulk Enrollment API tests are borrowed from and modified from upstream:
  - https://github.com/edx/edx-platform/blob/1990dc7/lms/djangoapps/bulk_enroll/tests/test_views.py
  - edx#15006

OmarIthawi commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(16) Appsembler API: Implement Test](https://trello.com/c/QMjvZFj6/1315-16-appsembler-api-implement-test)

bryanlandia commented on lms/djangoapps/appsembler_api/tests/test_bulk_enrollments.py:
```diff
@@ -0,0 +1,333 @@
+        res_json = json.loads(response.content)
+        self.assertEqual(res_json, expected)
+
+        # Check the outbox
```
... and then an additional method to exercise the condition where `'email_students'` is true ... and `len(mail.outbox) == 1`

bryanlandia commented on lms/djangoapps/appsembler_api/tests/test_create_user_account.py:
```diff
@@ -0,0 +1,64 @@
+class CreateUserAccountViewTests(TestCase):
+    def setUp(self):
+        self.url = reverse('create_user_account_api')
+        self.sample_user_date = {
```
nitpick:  Why `_date`?  Supposed to be `_data`?

bryanlandia commented on lms/djangoapps/appsembler_api/tests/test_permissions.py:
```diff
@@ -0,0 +1,40 @@
+    for member_name in dir(api_views):
+        member = getattr(api_views, member_name)
+        if isclass(member):
+            if member.__module__ == api_views.__name__:
```
Just curious... why this member.__module__ check?  Doesn't your for...in loop already take care of this restriction to `api_views`?

bryanlandia replied:
Oh,.... or does this prevent including imported classes?

OmarIthawi replied:
> Oh,.... or does this prevent including imported classes?

You got it right!

bryanlandia commented on lms/djangoapps/appsembler_api/tests/test_update_user_account.py:
```diff
@@ -0,0 +1,155 @@
+@patch('appsembler_api.views.UpdateUserAccount.authentication_classes', [])
+@patch('appsembler_api.views.UpdateUserAccount.permission_classes', [AllowAny])
+class CreateUserAccountViewTests(TestCase):
+    USER_FIELDS = [
```
Doesn't look like `USER_FIELDS` is used

OmarIthawi replied:
Terminated.

bryanlandia commented on lms/djangoapps/appsembler_api/views.py:
```diff
@@ -284,30 +284,31 @@ def post(self, request):
+            }, status=404)
+        except User.MultipleObjectsReturned:
+            return Response({
+                "lookup_error": "Two users have been found wiht the provided user_lookup",
```
nitpick:  Should be "found with"

OmarIthawi commented:
Thanks @bryanlandia for the review. I've addressed all of your comments, except for the `test_bulk_enrollments` module, since it's mostly copied from master and I don't feel like it's worth spending much effort on it given that this is already planned to be [retired in Hawthorne](https://trello.com/c/Ko5bm1EB/2726-move-the-customers-to-the-updated-hawthorne-bulk-enrollment-api).

Merging now.
